substrate/primitives/metadata-ir/src/types.rs

```diff
@@ -118,6 +120,87 @@ impl IntoPortable for RuntimeApiMethodParamMetadataIR {
 	}
 }
 
+/// Metadata of the the runtime query dispatch.
+#[derive(Clone, PartialEq, Eq, Encode, Debug)]
+pub struct RuntimeQueryIR<T: Form = MetaForm> {
+	/// The type implementing the runtime query dispatch.
+	pub ty: T::Type,
+	/// The query interfaces metadata.
+	pub interfaces: Vec<QueryInterfaceIR<T>>,
```

Comment:
Aren't pallet view functions scoped to certain pallets? So would it make sense for this information to be inside the `PalletMetadataIR`?

Reply:
This PR allows them to be defined only as part of pallets (because it is convenient for the initial implementation), but the requirement is to make them pallet agnostic, so they can be defined independently of pallets.

Reply:
What's the advantage/reason to replace Runtime APIs in the future with these? My understanding (which might miss something; I haven't read the background docs in a while not) is:

- We have "Runtime APIs" as a pallet agnostic set of APIs (called externally via eg `state_call` by using `RuntimeApiTraitName ++ _ ++ method_name` and SCALE encoding the args.
- We have here "Pallet View Functions", which are like Runtime APIs but are defined at the pallet level instead (and queried via a Runtime API call). These currently are accessed by using `twox_128(pallet_name) ++ twox_128("fn_name(fnarg_types) -> return_ty")` and SCALE encoding the args.

Both allow you to call functions and get the results back. In the future I'd expect both to become subscribable. The calling convetion of Pallet View Functions (hashing the type sig etc to query them) is different in order to enforce the type signature in the query (which means that changing the type signature of one would lead to old queries being invalid rather than being executed and complaining about wrongly encoded params or whatever). 

I guess overall I see them as working alongside Runtime APIs but at the pallet level, which is also why I've tended to push for making them consistent with Runtime APIs in terms of how they are talked about and called etc so far, but I feel like I am missing something because others seem less keen on this so far :D